src/ck-core/cklocation.C

```diff
@@ -2451,6 +2451,9 @@ void CkLocCache::recordEmigration(CmiUInt8 id, int pe)
 
   itr->second.pe = pe;
   itr->second.epoch++;
+
+  // inform listeners of the new location
+  notifyListeners(id, pe);
```

Comment:
One question is: why isn't this currently performed? If the PE that an element is emigrating from isn't notified of the emigration -- it could have unbuffered messages bound for that element. How is this avoided in the current system? Are the buffers periodically flushed? Is there an expectation that emigrating elements shouldn't have buffered messages?